File: src/main/java/com/ruby/pro203_exam/child/model/Child.java
package com.ruby.pro203_exam.child.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.UUID;

@Entity
@Table(name = "children")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Child {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "first_name", nullable = false, length = 100)
    private String firstName;

    @Column(name = "last_name", nullable = false, length = 100)
    private String lastName;

    @Column(name = "birth_date", nullable = false)
    private LocalDate birthDate;

    @Column(name = "group_name", length = 50)
    private String groupName;

    @Column(name = "group_id")
    private UUID groupId;

    // Link to health data
    @Column(name = "health_data_id")
    private UUID healthDataId;

    @Column(name = "kindergarten_id")
    private UUID kindergartenId;

    // Is currently in kindergarten?
    @Column(name = "checked_in", nullable = false)
    @Builder.Default
    private Boolean checkedIn = false;
}

File: frontend/services/types/child.ts
export interface ChildResponseDto {
    id: string;
    firstName: string;
    lastName: string;
    birthDate: string;
    profilePictureUrl?: string;
    kindergartenId: string;
    groupId?: string;
    groupName?: string;
    healthDataId?: string;
}

export interface CreateChildDto {
    firstName: string;
    lastName: string;
    birthDate: string;
    kindergartenId: string;
    groupId?: string;
}

export interface UpdateChildDto {
    firstName?: string;
    lastName?: string;
    birthDate?: string;
    groupId?: string;
    profilePictureUrl?: string;
}